### DISCUSSION TITLE
Customize Services to be able to auto create dns and LB i AWS

### DISCUSSION THREAD

**Discussion by Zodor** [arkmq-org/arkmq-org-broker-operator#554] [General] [answered]
Hi.

I want to be able to change the ClusterIP typ to LoadBalancer for the activemq 61616 port and wconsj 8161 ports to NodePort, because AWS Loadbalancer won't accept that the service-type is ClusterIP.

I have tried to change it myself, and it works like a charm (for some time). Then I think that the operator is changing it back to the original ClusterIP.

I have not idea how to make the port 61616 available from outside my cluster if I cannot accomplish this. We use the external-dns pods to automatically create route53 dns records and automatically create the LoadBalancer when the brokers go online. That can only be accomplished if we have the opportunity to change whatever annotation or spec that we want on the services for Artemis

**Comment by brusdev** [comment, marked as answer]
The operator reconciles resources created by itself so you need to create a new service with the same selector, in this way the operator will not reconcile your settings.

**Reply by Zodor** [reply]
Yes I have already done that, but that was not really the optimal solution. There should have been some options to add in the helmchart to configure the services and ingresses

**Reply by brusdev** [reply]
I see your point, it seems good to me having a filed that allow users to define how to expose an acceptor (ingress, route, node port), could you create a new issue?

**Reply by Zodor** [reply]
Sure. Here it is:
https://github.com/artemiscloud/activemq-artemis-operator/issues/557
